src/audio/volume/volume_hifi4_with_peakvol.c

```diff
@@ -470,10 +473,34 @@ static void vol_s16_to_s16(struct processing_module *mod, struct input_stream_bu
 			out_sample1 = AE_SLAI32S(out_sample1, 8);
 
 			/* store the output */
-			out_sample = AE_ROUND16X4F32SSYM(out_sample0, out_sample1);
+			out_sample = AE_ROUND16X4F32SSYM(out_sample0, out_sample0);
```

Comment:
Is this intentional? It seems we ignore the multiply result with "volume1". But can we then drop operations on output_sample1 altogether  and just produce out_samples0 ?

Reply:
Yes, definitely a mistake, must be a typo in working with code change options. Very good catch, I need to fix this. I'm guessing this caused the worse ramped audio quality in peakvol vs. gain that I was wondering.